### DISCUSSION TITLE
How to terminate server after start

### DISCUSSION THREAD

**Discussion by razetime** [triska/the-power-of-prolog#39] [Q&A]
running the query `server(Port).` starts the server, however the scryer prolog process runs even after Ctrl-Z, Ctrl-C and Ctrl-D are pressed. What is the intended way to stop the server?

**Comment by triska** [comment]
On OSX and Debian, I can press `Ctrl-z` to suspend the process and return to the terminal. On the terminal, I can enter:

<pre>
$ kill -9 %1
</pre>

to stop the server. Does this work for you?

**Reply by razetime** [reply]
Yes, on ubuntu this is what I was using. I was wondering if there was a simpler way to do so.

**Reply by triska** [reply]
Ideally, `socket_server_accept/4` can be interrupted with `Ctrl-c`. There is an existing Scryer Prolog issue for this: https://github.com/mthom/scryer-prolog/issues/485

Making this work would be an awesome contribution to Scryer Prolog!

**Comment by triska** [comment]
<s>Good news: The recently merged contribution **https://github.com/mthom/scryer-prolog/pull/3352** by @danilp-id addresses this!

If you are interested, could you please try it and see if it works for you?

Thank you a lot!</s>

**Update**: Ah, no! I now see this is currently confined to the HTTP library!

**Reply by triska** [reply]
**But now!!** Please check this out:

https://github.com/mthom/scryer-prolog/pull/3362
